### PR TITLE
Use go:embed instead of packr

### PR DESCRIPTION
**Pull request by makramkd**
Replace packr with Go 1.16's [embed](https://pkg.go.dev/embed) feature.

Here's a summary of the changes:
1. Use `go:embed` instead of `packr.Box` to embed the build assets of operator-ui into the chainlink binary.
2. In order to be able to use `go:embed`, we must have the assets available in the same folder as the module in which the `go:embed` directive is supplied. In this case, the assets must be available in `core/web`. Therefore, the webpack configuration was changed to output the compiled static assets (the HTML, JavaScript, and SVG files) in `core/web` instead of `operator_ui/dist`. 
3. As a consequence of doing (2), we need to update the chainlink dockerfile.

### CODE REVIEW

File: core/web/middleware.go [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
 	"strings"
 
 	"github.com/gin-gonic/gin"
-	"github.com/gobuffalo/packr"
+
 	"github.com/smartcontractkit/chainlink/core/logger"
 )
 
+// Go's new embed feature doesn't allow us to embed things outside of the current module.
+// To get around this, we basically need to copy over the assets to this module's directory
```
Comment (jkongie) [inline core/web/middleware.go:19]: You could try to output the yarn into the module instead perhaps?

Comment (archseer) [reply core/web/middleware.go:19]: Or a symlink could work?

File: go.mod [hunk 2]
```diff
@@ -100,9 +100,7 @@ require (
 	github.com/go-playground/validator/v10 v10.4.1 // indirect
 	github.com/go-stack/stack v1.8.0 // indirect
 	github.com/gobuffalo/envy v1.7.0 // indirect
-	github.com/gobuffalo/logger v1.0.0 // indirect
```
Comment (archseer) [inline go.mod:103]: Don't forget to remove it from the dependency list at the top of go.mod too

File: core/web/middleware.go (hunk 1 above)
Comment (jkongie) [reply core/web/middleware.go:19]: I believe you cannot use symlinks for security reasons

Comment (makramkd) [reply core/web/middleware.go:19]: If we output it to `core/web`, will that break things like running the frontend server separately? Or will that end up re-compiling assets anyway (without minifying probably, to aid in debugging)?

**Comment by jkongie** [comment]
I don’t believe so. It runs a node werver for local development, but if it
does rely on the same build path, you could make local dev load from the
original path

File: go.mod (hunk 2 above)
Comment (makramkd) [reply go.mod:103]: Done!

Comment (archseer) [reply go.mod:103]: You probably also need a `go mod tidy` now? There's still some references in `go.sum`

**Review by archseer — APPROVED** [review]

**Comment by archseer** [comment]
Oh, since the contracts are no longer embedded, I wonder if we could move https://github.com/smartcontractkit/chainlink/blob/80ce4718fd328d204634cd2c70ece51a1e8bd66a/.github/workflows/continuous-integration-workflow.yml#L64-L67 under the solidity workflow.

**Review by samsondav — APPROVED** [review]
Nice work

File: go.mod (hunk 2 above)
Comment (makramkd) [reply go.mod:103]: This is odd, I still see `packr` in the `go.sum` even after I run a `go mod tidy`. Even after manually deleting those entries in `go.sum` and re-calling `go mod tidy` it still adds them back.

File: .github/workflows/continuous-integration-workflow.yml [hunk 3]
```diff
@@ -61,10 +61,6 @@ jobs:
             go-mod-${{ env.CACHE_VERSION }}
       - name: Download Go vendor packages
         run: go mod download
-      - name: Compile all contracts
-        run: ./contracts/scripts/native_solc_compile_all
-      - name: Verify local solc binaries
-        run: ./tools/ci/check_solc_hashes
```
[the remark this answers is not in the document]
Comment (makramkd) [reply .github/workflows/continuous-integration-workflow.yml:67]: I can move this there, since we don't do it there.

File: go.mod (hunk 2 above)
Comment (makramkd) [reply go.mod:103]: Actually, another thing I noticed is that `go mod tidy` only removes some versions of a dependency from `go.sum` and not others. Not sure why that is. E.g `godirwalk`, `godotenv`, `go-internal`, and `logrus` still have entries in `go.sum`.

Comment (makramkd) [reply go.mod:103]: So despite this being somewhat annoying, I think this is alright. The `go.sum` file basically just has hashes of modules at certain versions, and it does this to verify dependencies it downloaded (as a security feature). It doesn't mean that the dep _will_ be downloaded though, so it's not dependency bloat. The `go.mod` file however is where deps are declared and that's the source of truth on what modules are required by the app.

**Review by archseer — APPROVED** [review]
Great work! 🎉

**Pull request merged**